lua/pac3/core/shared/movement.lua

```diff
@@ -145,6 +151,24 @@ pac.AddHook("Move", "custom_movement", function(ply, mv)
 
 	ply:SetJumpPower(self.JumpHeight)
 
+	if SERVER then
+		if allowMass:GetInt() == 1 then
+			ply:GetPhysicsObject():SetMass(math.Clamp(self.Mass, massLowerLimit:GetFloat(), massUpperLimit:GetFloat()))
+		end
+	end
+	
+	if (movementConvar:GetInt() == 1 or (movementConvar:GetInt() == -1 and hook.Run("PlayerNoClip", ply, true) == true)) and massDamageScale:GetInt() == 1 then
+		scale_mass = 85/math.Clamp(self.Mass, math.max(massLowerLimit:GetFloat(), 0.01), massUpperLimit:GetFloat())
+	else
+		scale_mass = 1
+	end
+	
+	pac.AddHook("EntityTakeDamage", "PAC3MassDamageScale", function(target, dmginfo)
+		if (target:IsPlayer() and dmginfo:IsDamageType(DMG_CRUSH or DMG_VEHICLE)) then
+				dmginfo:ScaleDamage(scale_mass)
+		end
+	end)
```

Comment:
This hook will be affecting every player rather than just the mass scaled player.